docs/source/notes/randomness.rst

```diff
@@ -2,29 +2,54 @@ Reproducibility
 ===============
 
 Completely reproducible results are not guaranteed across PyTorch releases,
-individual commits or different platforms. Furthermore, results need not be
+individual commits, or different platforms. Furthermore, results need not be
 reproducible between CPU and GPU executions, even when using identical seeds.
 
-However, in order to make computations deterministic on your specific problem on
-one specific platform and PyTorch release, there are a couple of steps to take.
+However, in order to make computations deterministic for your specific problem,
+platform, and PyTorch release, there are a couple of steps to take.
 
 There are two pseudorandom number generators involved in PyTorch, which you will
-need to seed manually to make runs reproducible. Furthermore, you should ensure
-that all other libraries your code relies on and which use random numbers also
-use a fixed seed.
+need to seed manually to make runs reproducible. Furthermore, you must configure
+PyTorch to only use deterministic algorithms. Finally, if your code relies on
+other libraries that use random numbers, make sure to give them a fixed seed.
 
-PyTorch
+Random number generation
 .......
 You can use :meth:`torch.manual_seed()` to seed the RNG for all devices (both
 CPU and CUDA)::
 
     import torch
     torch.manual_seed(0)
 
+Deterministic algorithms
+.......
+Some PyTorch operations use nondeterministic algorithms by default. A subset
+of them can be configured to use a deterministic implementation with
+:meth:`torch.set_deterministic()`. If an operation that does not currently have
+a deterministic implementation is called after this, a RuntimeError will be
+thrown so that you know to avoid that operation.
+
+.. code-block::
+
+    torch.set_deterministic(True)
+
+If an operation does not produce deterministic results after setting the
+deterministic flag, then the PyTorch team should make that operation throw
```

Comment:
Since this is user-facing, maybe instead of saying what the PyTorch team should do it can suggest the user file an issue so we can fix the problem?

Reply:
Ah, I see you're suggesting that later. I would still recommend cutting out the middle here. Maybe something more like: "If an operation fails to produce deterministic results after setting the deterministic flag, or you need a deterministic implementation of an operation, then please file an issue at <link>."

?